sanic/request.py

```diff
@@ -355,19 +355,38 @@ def _get_address(self):
 
     @property
     def remote_addr(self):
-        """Attempt to return the original client ip based on X-Forwarded-For.
+        """Attempt to return the original client ip based on X-Forwarded-For
+        or X-Real-IP. If HTTP headers are unavailable or untrusted, returns
+        an empty string.
 
         :return: original client ip.
         """
         if not hasattr(self, "_remote_addr"):
-            forwarded_for = self.headers.get("X-Forwarded-For", "").split(",")
-            remote_addrs = [
-                addr
-                for addr in [addr.strip() for addr in forwarded_for]
-                if addr
-            ]
-            if len(remote_addrs) > 0:
-                self._remote_addr = remote_addrs[0]
+            if self.app.config.PROXIES_COUNT == 0:
```

Comment:
Since this if and the last else block is doing `self._remote_addr = ""`, we can do that outside the `if` block and reduce the number if `if-else` branching.

Reply:
In this case I'll have to use more indentation levels or prepend every expression with `if self.app.config.PROXIES_COUNT is not None and self.app.config.PROXIES_COUNT > 0 and [...]` that is more complicated

(Or I'm too stupid and can't see a more obvious and beautiful way to do this)

Reply:
It's small change that isn't super important. 

I was suggesting to remove the 
```python
if self.app.config.PROXIES_COUNT == 0:
    self._remote_addr = ""
```
and 
```
else:
    self._remote_addr = ""
```
and instead merge them as 
```python
if not hasattr(self, "_remote_addr"):
    self._remote_addr = ""
```

Reply:
@andreymal would you be ok with making the above change?

Reply:
@harshanarayana when you get a chance, could you look at this again?

Reply:
@sjsadowski  @andreymal 
```python
        if not hasattr(self, "_remote_addr"):
            setattr(self, "_remote_addr", "")

        if self.app.config.PROXIES_COUNT != 0:
            if self.app.config.REAL_IP_HEADER and self.headers.get(self.app.config.REAL_IP_HEADER):
                self._remote_addr = self.headers[self.app.config.REAL_IP_HEADER]
            elif self.app.config.FORWARDED_FOR_HEADER:
                forwarded_for = self.headers.get(self.app.config.FORWARDED_FOR_HEADER, "").split(",")
                remote_addrs = [addr for addr in [addr.strip() for addr in forwarded_for] if addr]
                if self.app.config.PROXIES_COUNT == -1:
                    self._remote_addr = remote_addrs[0]
                elif len(remote_addrs) >= self.app.config.PROXIES_COUNT:
                    self._remote_addr = remote_addrs[-self.app.config.PROXIES_COUNT]

        return self._remote_addr
```

This is what I had in mind. But then again this is more of a cosmetic change.

Reply:
@harshanarayana I'm going to approve as-is, without this change since it's cosmetic.